src/transformers/models/splinter/tokenization_splinter.py

```diff
@@ -0,0 +1,531 @@
+# coding=utf-8
+# Copyright Ori Ram, Yuval Kirstain, Jonathan Berant, Amir Globerson, Omer Levy and The HuggingFace Inc. team.
+# All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tokenization classes for Splinter."""
+
+import collections
+import os
+import unicodedata
+from typing import List, Optional, Tuple
+
+from ...tokenization_utils import PreTrainedTokenizer, _is_control, _is_punctuation, _is_whitespace
+from ...utils import logging
+
+
+logger = logging.get_logger(__name__)
+
+VOCAB_FILES_NAMES = {"vocab_file": "vocab.txt"}
+
+PRETRAINED_VOCAB_FILES_MAP = {
+    "vocab_file": {
+        "tau/splinter-base": "https://huggingface.co/tau/splinter-base/resolve/main/vocab.txt",
+        "tau/splinter-base-qass": "https://huggingface.co/tau/splinter-base-qass/resolve/main/vocab.txt",
+        "tau/splinter-large": "https://huggingface.co/tau/splinter-large/resolve/main/vocab.txt",
+        "tau/splinter-large-qass": "https://huggingface.co/tau/splinter-large-qass/resolve/main/vocab.txt",
+    }
+}
+
+PRETRAINED_POSITIONAL_EMBEDDINGS_SIZES = {
+    "tau/splinter-base": 512,
+    "tau/splinter-base-qass": 512,
+    "tau/splinter-large": 512,
+    "tau/splinter-large-qass": 512,
+}
+
+PRETRAINED_INIT_CONFIGURATION = {
+    "tau/splinter-base": {"do_lower_case": False},
+    "tau/splinter-base-qass": {"do_lower_case": False},
+    "tau/splinter-large": {"do_lower_case": False},
+    "tau/splinter-large-qass": {"do_lower_case": False},
+}
+
+
+def load_vocab(vocab_file):
+    """Loads a vocabulary file into a dictionary."""
+    vocab = collections.OrderedDict()
+    with open(vocab_file, "r", encoding="utf-8") as reader:
+        tokens = reader.readlines()
+    for index, token in enumerate(tokens):
+        token = token.rstrip("\n")
+        vocab[token] = index
+    return vocab
+
+
+def whitespace_tokenize(text):
+    """Runs basic whitespace cleaning and splitting on a piece of text."""
+    text = text.strip()
+    if not text:
+        return []
+    tokens = text.split()
+    return tokens
+
+
+class SplinterTokenizer(PreTrainedTokenizer):
```

Comment:
Is this tokenizer any different from BERT or is it the exact same?

Reply:
Hi @LysandreJik,
Same as above - The tokenizer is indeed different (see `build_inputs_with_special_tokens` )